### PR TITLE
Deprecate @higherkind & @extension for TupleN

### PR DESCRIPTION
**Pull request by nomisRev**
- [x] Show
- [x] Eq
- [x] Hash
- [x] Order
- [x] Semigroup for Tuple2
- [x] Monoid for Tuple2
- [x] Deprecate Monad hierarchy for Tuple2
- [x] Deprecate Functor hierarchy for Tuple2
- [x] Deprecate Apply hierarchy for Tuple2
- [x] Deprecate Applicative hierarchy for Tuple2
- [x] Deprecate Comonad hierarchy for Tuple2
- [x] Deprecate Foldable hierarchy for Tuple2
- [x] Deprecate BiFunctor hierarchy for Tuple2
- [x] Deprecate BiFoldable hierarchy for Tuple2
- [x] Deprecate Traverse hierarchy for Tuple2
- [x] Deprecate BiTraverse hierarchy for Tuple2

### CODE REVIEW

File: arrow-core-data/src/main/kotlin/arrow/core/Tuple10.kt [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    HB.run {
+      HC.run {
+        HD.run {
+          HE.run {
+            HF.run {
+              HG.run {
+                HH.run {
+                  HI.run {
+                    HJ.run {
+                      a.hashWithSalt(
```
Comment (raulraja) [inline arrow-core-data/src/main/kotlin/arrow/core/Tuple10.kt:153]: LOL this must have been painful!

File: arrow-core/src/main/kotlin/arrow/core/extensions/tuple2/bitraverse/Tuple2Bitraverse.kt [hunk 2]
```diff
@@ -50,14 +43,7 @@ fun <G, A, B, C, D> Kind<Kind<ForTuple2, A>, B>.bitraverse(
...
 )
-@Deprecated(
-  "@extension kinded projected functions are deprecated",
-  ReplaceWith(
-  "bisequence(arg1)",
-  "arrow.core.bisequence"
-  ),
-  DeprecationLevel.WARNING
-)
+@Deprecated("Applicative typeclasses is deprecated. Use concrete methods on Pair")
```
Comment (raulraja) [inline arrow-core/src/main/kotlin/arrow/core/extensions/tuple2/bitraverse/Tuple2Bitraverse.kt:46]: nit: `are` vs `is` or `typeclass` vs `typeclasses`

File: arrow-core/src/main/kotlin/arrow/core/extensions/tuple2/order/Tuple2Order.kt [hunk 3]
```diff
@@ -18,10 +18,10 @@ import kotlin.jvm.JvmName
   "UNUSED_PARAMETER"
 )
 @Deprecated(
-  "@extension kinded projected functions are deprecated",
+  "Tuple2 is deprecated in favor of Kotlin's Pair. Use Par functionality.",
```
Comment (raulraja) [inline arrow-core/src/main/kotlin/arrow/core/extensions/tuple2/order/Tuple2Order.kt:21]:
```suggestion
  "Tuple2 is deprecated in favor of Kotlin's Pair. Use Pair functionality.",
```

**Review by raulraja — APPROVED** [review]
Thanks @nomisRev !

**Review by i-walker — APPROVED** [review]
🙌 Thanks @nomisRev !

File: arrow-core/src/main/kotlin/arrow/core/extensions/tuple2/bitraverse/Tuple2Bitraverse.kt (hunk 2 above)
Comment (nomisRev) [reply arrow-core/src/main/kotlin/arrow/core/extensions/tuple2/bitraverse/Tuple2Bitraverse.kt:46]:
```suggestion
@Deprecated("Applicative typeclass is deprecated. Use concrete methods on Pair")
```

**Pull request merged**